Automattic/jetpack
# AI Assistant: Add List and List Item inline extensions

Fixes #37478
Fixes #37563

Depends on D150004-code

## Proposed changes:

* Adds the List and List Item extensions to the unreleased inline extensions list
* Adds HTML fixes to ensure the content is streamed correctly

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Apply D150004-code on your sandbox (if not already deployed) and sandbox the public API
* Go to the block editor and create a List. Tip: use the AI Assistant to generate one, like "list the world's continents sorted by area, informing the area and a fun fact about each one". This should generate a list with sublists.
* Use the inline extension on the List and List Item blocks
* Check that the input is shown at the right place and the extension works like in the previous extensions.


https://github.com/Automattic/jetpack/assets/8486249/60e1c8ab-7f47-4961-9730-a90643c4b38a
